### ISSUE TITLE
Graphql and Lucene problem

### ISSUE THREAD

**Issue by vitalybrandes** [OrchardCMS/OrchardCore#5782]
Once Lucene features graphql enabled together the Graphql Shcema not available

Nuget ver 

![1](https://user-images.githubusercontent.com/61165619/77320015-7c48a600-6d18-11ea-97c8-003adc07e801.PNG)
![2](https://user-images.githubusercontent.com/61165619/77320016-7ce13c80-6d18-11ea-90f4-95da27287ad5.PNG)
![3](https://user-images.githubusercontent.com/61165619/77320018-7d79d300-6d18-11ea-990d-c891ed17495d.PNG)

Version="1.0.0-rc1-10004"

**Comment by Skrypt** [comment]
Too old version for affirming it's an issue we can repro. Upgrade to latest Myget versions.

**Comment by vitalybrandes** [comment]
Get the latest from MyGet,
exactly the same,
once graphql and Lucene activated, no Schema available

**Comment by vitalybrandes** [comment]
> 
> 
> Too old version for affirming it's an issue we can repro. Upgrade to latest Myget versions.

Get the latest from MyGet,
exactly the same,
once graphql and Lucene activated, no Schema available

**Comment by Skrypt** [comment]
Have you created an index?

**Comment by vitalybrandes** [comment]
> Have you created an index?

Nope, I just enable the module.
Why do I need to index if I would.like to work with graphql first on something else? , Just saying...

**Comment by vitalybrandes** [comment]
Did that now,
same..
Found more issue,
I have to content Item, Customer and Machine, Machine is Content Picker Filed in Customer.
After Lucene activated, i cannot add any Machine 
![1](https://user-images.githubusercontent.com/61165619/77364126-986e3680-6d5c-11ea-908e-17ba99fec14c.PNG)

**Comment by deanmarcussen** [comment]
Having difficulty reproducing this.
With a blank recipe, enabling GraphQL, with no lucene indexes, works fine.

@vitalybrandes can you check your console/ and or App_Data logs?

It maybe you are experiencing this issue https://github.com/OrchardCMS/OrchardCore/issues/5689

**Comment by vitalybrandes** [comment]
> 
> 
> Having difficulty reproducing this.
> With a blank recipe, enabling GraphQL, with no lucene indexes, works fine.
> 
> @vitalybrandes can you check your console/ and or App_Data logs?
> 
> It maybe you are experiencing this issue #5689

I am using my own custom theme with custom recipie based on SAAS...
This is my recipie
```
     "OrchardCore.Admin",
        "OrchardCore.AdminMenu",
        "OrchardCore.Diagnostics",
        "OrchardCore.DynamicCache",
        "OrchardCore.HomeRoute",
        "OrchardCore.Localization",
        "OrchardCore.Features",
        "OrchardCore.Navigation",
        "OrchardCore.Menu",
        "OrchardCore.Recipes",
        "OrchardCore.Resources",
        "OrchardCore.Roles",
        "OrchardCore.Email",
        "OrchardCore.Settings",
        "OrchardCore.Tenants",
        "OrchardCore.Themes",
        "OrchardCore.Users",
```

Created number of content item, "sharing" them via graphql works ok, 
Once wanted to add Lucene searches , the Schema of graphql disappear.
I think it is some issue with the order of module loading.

Queries and Indexes module enabled!

**Comment by deanmarcussen** [comment]
I doubt there is any issue with module ordering, that's all well managed by the dependency graph.

I say again...
@vitalybrandes can you check your console/ and or App_Data logs?

**Comment by vitalybrandes** [comment]
Well, Update all nugets again, rebuild everything + fixed some order in Recipe, now its working.
Closing for now.

**Comment by mkgn** [comment]
Thought to reopen since it just happened in the latest version. See the below steps which caused the error.

1) Brand new Orchard setup using SQLite
2) Enabled Lucene. made some queries, tested and saved.
3) Enabled GraphQL went to editor and I could see the items in explorer. However queries were not listed since I haven't added a schema to lucene queries
4) Went back to lucene qeuries and added a schema like;

    {
         "type": "ContentItem/Test"
    }

5) Came back to GraphQL and explorer is now empty
6) Tried enable/disable lucene/graphql modules but issue remained.
7) Noticed below console error consistently appearing when I go to GraphQL section

![image](https://user-images.githubusercontent.com/8392595/209514262-542b677e-aecf-4f61-9e53-6a226080b8c0.png)

8) Noticed some localstorage data related to GraphQL. Cleared those, cleaned the solution, restarted the application but problem still remains. No exceptions related to this in logs.

Only thing I can think of is something happened at Step 4. But now even if i removed the schema and try it doesn't work.

**Comment by mkgn** [comment]
> Thought to reopen since it just happened in the latest version. See the below steps which caused the error.
> 
> 1. Brand new Orchard setup using SQLite
> 2. Enabled Lucene. made some queries, tested and saved.
> 3. Enabled GraphQL went to editor and I could see the items in explorer. However queries were not listed since I haven't added a schema to lucene queries
> 4. Went back to lucene qeuries and added a schema like;
>    {
>    "type": "ContentItem/Test"
>    }
> 5. Came back to GraphQL and explorer is now empty
> 6. Tried enable/disable lucene/graphql modules but issue remained.
> 7. Noticed below console error consistently appearing when I go to GraphQL section
> 
> ![image](https://user-images.githubusercontent.com/8392595/209514262-542b677e-aecf-4f61-9e53-6a226080b8c0.png)
> 
> 8. Noticed some localstorage data related to GraphQL. Cleared those, cleaned the solution, restarted the application but problem still remains. No exceptions related to this in logs.
> 
> Only thing I can think of is something happened at Step 4. But now even if i removed the schema and try it doesn't work.

Related to the above error, noticed that there is an API call being made when going to GraphQL section. it posts below data and it doesn't seem to have a "data" property as the error says. May be this will help?

{
  "query": "query IntrospectionQuery {  __schema {        queryType { name }    mutationType { name }    subscriptionType { name }    types {      ...FullType    }    directives {      name      description            locations      args {        ...InputValue      }    }  }}\nfragment FullType on __Type {  kind  name  description    fields(includeDeprecated: true) {    name    description    args {      ...InputValue    }    type {      ...TypeRef    }    isDeprecated    deprecationReason  }  inputFields {    ...InputValue  }  interfaces {    ...TypeRef  }  enumValues(includeDeprecated: true) {    name    description    isDeprecated    deprecationReason  }  possibleTypes {    ...TypeRef  }}\nfragment InputValue on __InputValue {  name  description  type { ...TypeRef }  defaultValue    }\nfragment TypeRef on __Type {  kind  name  ofType {    kind    name    ofType {      kind      name      ofType {        kind        name        ofType {          kind          name          ofType {            kind            name            ofType {              kind              name              ofType {                kind                name              }            }          }        }      }    }  }}\n  "
}

**Comment by mkgn** [comment]
This error seems to have a fix #5689. However the schema I entered in my query is simply the type of the content.

    {
        "type": "ContentItem/Test"
    }

![image](https://user-images.githubusercontent.com/8392595/209517165-66f5bd6b-de97-42a2-9997-4aa286c0194d.png)


my content type "technical name" is Test. So, I can't figure out what I am doing wrong here.
